### PR TITLE
fix: allow populating subfields for field groups and repeaters

### PR DESCRIPTION
**Pull request by jperasmus**
Populating subfields for both Field Groups and Repeater fields should now work with either `populate: true` or explicitly setting it with 
```
populate: [{
  field: 'repeater-or-fieldgroup',
  subFields: ['subfield1', 'subfield2'] // this can be a string or field object
}]
```

### CODE REVIEW

File: packages/utils/src/index.ts [hunk 1]
```diff
@@ -508,58 +508,48 @@ export const processReferencesForCF = curry(
...
+      const { field, populate, subFields } = opt
       const val = get(document, field)
 
       const processRefs = processReferencesForCF(firestoreService, {
-        populate: populateAllTheThings ? true : populate
+        populate: populateAllTheThings
+          ? true
+          : Array.isArray(subFields)
+          ? { populate: subFields }
+          : populate
```
Comment (jordanch) [inline packages/utils/src/index.ts]: nah nah nah!!! ternary ternary!!! my braaaaaain

File: packages/utils/src/index.ts [hunk 2]
```diff
@@ -508,58 +508,48 @@ export const processReferencesForCF = curry(
...
+          return Promise.all(docs.map(async doc => processRefs(doc)))
+        }
+
+        return processRefs(snapshot.data())
+      }
+
       if (Array.isArray(val)) {
         const fieldValue = await Promise.all(
           val.map(async innerRef => {
             if (isRefLike(innerRef)) {
```
Comment (jordanch) [inline packages/utils/src/index.ts]: cool.

File: packages/utils/src/index.ts [hunk 3]
```diff
@@ -508,58 +508,48 @@ export const processReferencesForCF = curry(
...
       const processRefs = processReferencesForCF(firestoreService, {
-        populate: populateAllTheThings ? true : populate
+        populate: populateAllTheThings
+          ? true
+          : Array.isArray(subFields)
+          ? { populate: subFields }
+          : populate
       })
 
+      const processRef = async (ref: any) => {
```
Comment (jordanch) [inline packages/utils/src/index.ts]: Can we not determine the different firestore reference types and logical OR them as types?

File: packages/utils/src/index.ts (hunk 1 above)
Comment (jperasmus) [reply packages/utils/src/index.ts]: ya, haha, don't like nested ternaries, but actually ended up being the easiest fix. Went down the rabbit hole of trying to handle subFields inside each type of value check and then realized I could just handle it recursively if the subfields become the next level's main `populate` fields.

File: packages/utils/src/index.ts (hunk 3 above)
Comment (jperasmus) [reply packages/utils/src/index.ts]: nah, unfortunately not. They are all just seen as Firestore Document References. To which type of document it refers, it has no idea. Could be a permissions document, an image document or a user, etc.

**Pull request merged**